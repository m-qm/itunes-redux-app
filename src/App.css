@import "~antd/dist/antd.css";

.App {
  text-align: center;
}

.App-header {
  height: 60px;
  padding: 5px;
  margin: 10px;
  background: #656565;
}

.App-title {
  font-size: 2.4em;
  color: white;
}

.App-body {
  margin-top: 20px;
}

@media (max-width: 650px) {
  .App-header {
    height: 45px;
  }

  .App-title {
    font-size: 1.7em;
  }
}

@media (max-width: 468px) {
  .App-header {
    height: 40px;
  }

  .App-title {
    font-size: 1.4em;
  }
}
