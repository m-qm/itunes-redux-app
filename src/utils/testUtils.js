import React from "react";

export const mockComponent = (name, props) => {
  const { children, ...restOfProps } = props;
  return (
    <mocked name={name} props={restOfProps}>
      {children}
    </mocked>
  );
};
