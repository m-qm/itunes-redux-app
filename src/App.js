import React, { Component } from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route } from "react-router-dom";
import "./App.css";

import configureStore from "./config/store";
import Search from "./components/Search";
import Detail from "./components/Detail";

const store = configureStore();

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <header className="App-header">
            <h1 className="App-title">Find your music</h1>
          </header>
          <div className="App-body">
            <Router>
              <div>
                <Route exact path="/" component={Search} />
                <Route path="/detail" component={Detail} />
              </div>
            </Router>
          </div>
        </div>
      </Provider>
    );
  }
}

export default App;
