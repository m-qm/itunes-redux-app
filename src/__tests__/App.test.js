import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { mockComponent } from "../utils/testUtils";

import App from "../App";

jest.mock("../components/Search", () => {
  return props => mockComponent("Search", props);
});

describe("App component", () => {
  const initialEntries = ["/"];
  const component = (
    <MemoryRouter initialEntries={initialEntries}>
      <App />
    </MemoryRouter>
  );

  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Snapshot matchs", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
