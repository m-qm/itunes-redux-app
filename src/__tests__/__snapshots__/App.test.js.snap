// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot matchs 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Find your music
    </h1>
  </header>
  <div
    className="App-body"
  >
    <div>
      <mocked
        name="Search"
        props={
          Object {
            "history": Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            },
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "match": Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            },
            "staticContext": undefined,
          }
        }
      />
    </div>
  </div>
</div>
`;
